### PR TITLE
Adds multiple package managers to `link` script

### PR DESCRIPTION
**Pull request by TheSonOfThomp**
Previously the `lg link` script would only install & link packages using yarn. Now it will check the destination app's package manager (via lock file) and link packages using the correct package manager.

Additionally, this PR tests for many `link` utils

### CODE REVIEW

**Review by bruugey — COMMENTED** [review]
Needs a changeset

File: tools/link/src/utils/createLinkFrom.spec.ts [hunk 1]
```diff
@@ -0,0 +1,49 @@
...
+      scopeName: '@example',
+      scopePath: 'scope',
+      packageName: 'test-package',
+    });
+
+    expect(spawnSpy).toHaveBeenCalledWith(
+      'npm',
+      ['link'],
+      expect.objectContaining({
+        cwd: expect.stringContaining('tmp/scope/test-package'),
```
Comment (bruugey) [inline tools/link/src/utils/createLinkFrom.spec.ts:45]: should this be @example/test-pacakge?

File: tools/link/src/utils/install.spec.ts [hunk 2]
```diff
@@ -0,0 +1,48 @@
+import { ChildProcess } from 'child_process';
+import xSpawn from 'cross-spawn';
+import fsx from 'fs-extra';
+
+// const withLocalTmpDir = require('with-local-tmp-dir')
```
Comment (bruugey) [inline tools/link/src/utils/install.spec.ts:5]: nit rm?

File: tools/link/src/utils/install.ts [hunk 3]
```diff
@@ -1,7 +1,13 @@
...
+import { SupportedPackageManager } from '@lg-tools/meta/src/getPackageManager';
 import { spawn } from 'cross-spawn';
+import fsx from 'fs-extra';
 
+/**
+ * @deprecated use `installPackages` instead
+ */
 export function yarnInstall(path: string) {
-  return new Promise((resolve, reject) => {
+  return new Promise(resolve => {
```
Comment (bruugey) [inline tools/link/src/utils/install.ts:10]: should we just rm this instead of deprecating it?

File: tools/meta/src/getPackageManager.ts [hunk 4]
```diff
@@ -1,11 +1,19 @@
 import fse from 'fs-extra';
+
+export type SupportedPackageManager = `npm` | `yarn` | `pnpm`;
+
 /** Reads the lock file to determine which package manager to use */
-export function getPackageManager(appPath: string): `npm` | `yarn` {
+export function getPackageManager(
+  appPath: string,
+  defaultPackageManager: SupportedPackageManager = 'npm',
```
Comment (bruugey) [inline tools/meta/src/getPackageManager.ts:8]: why is defaultPackageManager an arg?

Comment (TheSonOfThomp) [reply tools/meta/src/getPackageManager.ts:8]: Doesn't need to be, but I wanted to add the flexibility of preferring some other default if there's no lock file found

File: tools/link/src/utils/install.ts (hunk 3 above)
Comment (TheSonOfThomp) [reply tools/link/src/utils/install.ts:10]: Probably 👍

File: tools/link/src/utils/createLinkFrom.spec.ts (hunk 1 above)
Comment (TheSonOfThomp) [reply tools/link/src/utils/createLinkFrom.spec.ts:45]: No, `cwd` is the directory where the command is run from. We expect that to be the `test-package` dir

Comment (TheSonOfThomp) [reply tools/link/src/utils/createLinkFrom.spec.ts:45]: No, `cwd` is the directory where the command is run from. We expect that to be the `test-package` dir

**Review by bruugey — COMMENTED** [review]
How did you test this?

**Review by bruugey — APPROVED** [review]

**Pull request merged**